### PR TITLE
Support mTLS auth for quota app

### PR DESCRIPTION
**Pull request by Pooja-08**

### CODE REVIEW

File: helm-charts/interoperator/conf/settings.yaml [hunk 1]
```diff
@@ -100,6 +100,10 @@ defaults: &defaults
     enabled: {{ .Values.broker.quota.enabled }}
     oauthDomain: {{ .Values.broker.quota.oauthDomain }}
     serviceDomain: {{ .Values.broker.quota.serviceDomain }}
+    mtls:
+      enabled: {{ .Values.broker.quota.mtls.enabled }}
+      cert_secret: {{ .Values.broker.quota.mtls.cert_secret }}
```
Comment (anoopjb) [inline helm-charts/interoperator/conf/settings.yaml:105]:
```suggestion
```
The `cert_secret` is not used inside the pod. So I feel we don't need cert_secret in settings.yaml.

**Review by anoopjb — APPROVED** [review]
LGTM

**Comment by Pooja-08** [comment]
Please review "https://github.wdf.sap.corp/servicefabrik/interoperator-deployment-component/pull/46" for the document.

File: broker/core/utils/src/HttpClient.js [hunk 2]
```diff
@@ -50,10 +51,27 @@ class HttpClient {
...
+      const cert = fs.readFileSync('/etc/certs/cert', 'utf8');
+      const key = fs.readFileSync('/etc/certs/key', 'utf8');
+      if (_.isEmpty(cert) || _.isEmpty(key)) {
+        logger.error('Either certificate or key required for quota app authentication, is missing');
+        throw new InternalServerError('Missing certificate or key required to authenticate quota API call');
+      }
+      httpsAgent = new https.Agent({
+        rejectUnauthorized: reject_unauthorized,
+        cert: cert,
+        key: key
```
Comment (saud89) [inline broker/core/utils/src/HttpClient.js:67]: Hi @Pooja-08, please add timeout option here... probably as enhancement in next pr

**Review by saud89 — APPROVED** [review]

File: broker/core/utils/src/HttpClient.js (hunk 2 above)
Comment (Pooja-08) [reply broker/core/utils/src/HttpClient.js:67]: sure.. will do so

**Pull request merged**